### PR TITLE
Fix namespace conflict with hypre [namespace-conflict-bugfix]

### LINKED ISSUE #225: Namespace conflict with hypre
In v3.3 and newer (including git-master as of 06/22/17), the use of HYPRE_PARCSR in the enum Type member of the Operator class (linalg/operator.hpp) conflicts with its use as a #define'd constant in HYPRE.h within the hypre library.  Reproducible by using

#include "HYPRE.h"
#include "mfem.hpp"

in the headers (although the opposite order does work fine).  Suggested fix is to use a different keyword here that doesn't conflict with the hypre namespace.

### PR DESCRIPTION
**Pull request by tzanio**
Rename `Operator::HYPRE_PARCSR` to `Operator::MFEM_HYPRE_PARCSR` to resolve #225.

### CODE REVIEW

File: examples/petsc/ex5p.cpp [hunk 1]
```diff
@@ -210,7 +210,7 @@ int main(int argc, char *argv[])
    PetscParMatrix *pM = NULL, *pB = NULL, *pBT = NULL;
    HypreParMatrix *M = NULL, *B = NULL, *BT = NULL;
    Operator::Type tid =
-      !use_petsc ? Operator::HYPRE_PARCSR :
+      !use_petsc ? Operator::MFEM_HYPRE_PARCSR :
```
Comment (v-dobrev) [inline examples/petsc/ex5p.cpp]: The `MFEM_` prefix is redundant  - `Operator` is already in the `mfem` namespace - so I prefer to use something like `Hypre_ParCSR` where not all characters are capitalized.

Comment (tzanio) [reply examples/petsc/ex5p.cpp]: I'm OK with that, as long as _hypre_ doesn't decide to define a public macro for this too. I agree that it is sub-optimal, but that's the reason for the `MFEM_` prefix.

File: linalg/hypre_parcsr.hpp [hunk 2]
```diff
@@ -9,8 +9,8 @@
 // terms of the GNU Lesser General Public License (as published by the Free
 // Software Foundation) version 2.1 dated February 1999.
 
-#ifndef MFEM_HYPRE_PARCSR
-#define MFEM_HYPRE_PARCSR
+#ifndef MFEM_HYPRE_PARCSR_FILE
+#define MFEM_HYPRE_PARCSR_FILE
```
Comment (v-dobrev) [inline linalg/hypre_parcsr.hpp]: I think the `_FILE` suffix should be replaced with `_HPP`.

Comment (tzanio) [reply linalg/hypre_parcsr.hpp]: ok

File: examples/petsc/ex5p.cpp (hunk 1 above)
Comment (ScottMacLachlan) [reply examples/petsc/ex5p.cpp]: I agree with Tzanio - that the change is fine, so long as hypre doesn't change as well.  I prefer the MFEM_ prefix myself...

**Pull request merged**